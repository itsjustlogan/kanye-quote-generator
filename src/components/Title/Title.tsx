import React from 'react'

interface Props {
  title: string
}

export default function Title({ title }: Props) {
  return (
    <h1
      data-testid="title"
      aria-label="app title"
      className="text-center py-6 text-green-200 bg-red-200 font-3xl xl:font-4xl"
    >
      {title}
    </h1>
  )
}
