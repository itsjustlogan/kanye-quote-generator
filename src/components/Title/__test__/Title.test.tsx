import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Title from '../Title'

describe('Title Component', () => {
  afterEach(() => {
    cleanup()
  })

  it('should render correctly', () => {
    expect(Title).toBeTruthy()

    const { asFragment, rerender } = render(<Title title="app title" />)

    expect(asFragment().textContent).toContain('app title')

    rerender(<Title title="new app title" />)
    expect(asFragment().textContent).toContain('new app title')
  })

  it('should match screenshot', () => {
    const { asFragment } = render(<Title title="very deep quote" />)

    expect(asFragment()).toMatchSnapshot()
  })
})
