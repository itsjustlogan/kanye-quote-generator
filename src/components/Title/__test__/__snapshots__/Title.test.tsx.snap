// Vitest Snapshot v1

exports[`Title Component > should match screenshot 1`] = `
<DocumentFragment>
  <h1
    aria-label="app title"
    class="text-center py-6 text-green-200 bg-red-200 font-3xl xl:font-4xl"
    data-testid="title"
  >
    very deep quote
  </h1>
</DocumentFragment>
`;
